source/ui/BioXAS/BioXASControlButton.cpp

```diff
@@ -0,0 +1,69 @@
+#include "BioXASControlButton.h"
+
+BioXASControlButton::BioXASControlButton(AMControl *control, QWidget *parent) :
+	AMToolButton(parent)
+{
+	// Initialize class variables.
+
+	control_ = 0;
+
+	greenValueSet_ = false;
+	greenValue_ = 0;
+
+	// Current settings.
+
+	setControl(control);
+
+	refresh();
+}
+
+BioXASControlButton::~BioXASControlButton()
+{
+
+}
+
+void BioXASControlButton::refresh()
+{
+	QColor color = QColor();
+
+	if (control_) {
+		if (greenValueSet_ && control_->isConnected() && qFuzzyCompare(control_->value(), greenValue_))
+			color = QColor(Qt::green);
+		else if (greenValueSet_ && control_->isConnected())
+			color = QColor(Qt::red);
+		else if (greenValueSet_)
+			color = QColor(Qt::yellow);
```

Comment:
It displays the 'default' color--the button has no color. Just looks like a regular button.